pkg/agent/manager/manager.go

```diff
@@ -113,13 +121,23 @@ func (m *manager) Run(ctx context.Context) error {
 		m.runSVIDObserver,
 		m.runBundleObserver,
 		m.svid.Run)
-	if err != nil && err != context.Canceled {
-		m.c.Log.WithError(err).Error("cache manager crashed")
+
+	switch {
```

Comment:
This switch seems a little unwieldy. How would you feel about the following?
```suggestion
	switch {
	case err == nil || err == context.Canceled:
		m.c.Log.Info("Cache manager stopped")
		return nil
	case shouldAgentReattest(err):
		m.c.Log.WithError(err).Warn("Agent needs to re-attest; removing SVID and shutting down")
		... Rest of the delete logic
		return err
	default:
		m.c.Log.WithError(err).Error("cache manager crashed")
		return err
	}
```

Reply:
Thanks for that snippet!